content/blog/2019/corda-simple-broadcast-transaction/index.md

```diff
@@ -0,0 +1,71 @@
+
+```kotlin
+@InitiatedBy(BroadcastTransactionFlow::class)
+class BroadcastTransactionResponder(private val session: FlowSession) : FlowLogic<Unit>() {
+
+  @Suspendable
+  override fun call() {
+    subFlow(ReceiveTransactionFlow(session, statesToRecord = StatesToRecord.ALL_VISIBLE))
+  }
+}
+```
+
+There is only a single line inside of `call`. So, due to nothing else being there, it must be important. `ReceiveTransactionFlow` is the counterpart of `SendTransactionFlow` who receives and persists the transaction sent to it. Furthermore, the `statesToRecord` property determines which states from the transaction should be stored in the vault. Since this is a broadcast (possibly for regulatory reasons) the contents of the transaction are going to be important. To achieve this, `StatesToRecord.ALL_VISIBLE` is used telling `ReceiveTransactionFlow` to record every state contained within the transaction. If, depending on your use-case, you do not require all of the transaction's states, then you can use `ONLY_RELEVANT` or even `NONE`.
+
+The last important line of code in these snippets is the `@InitiatingFlow` that `BroadcastTransactionFlow` is annotated with. This allows the flow to work without requiring its caller to create sessions and before passing them into the flow. The flow will handle that. Logically this makes sense since most of the time you will not have initiated sessions with parties that don't already have the transaction stored. Thanks to this, using this flow from inside another is nice and simple. For example, it could be added at the end of a flow:
+
+```kotlin
+subFlow(FinalityFlow(stx, sessions)).also {
+  // sends to everyone in the network
+  val broadcastToParties =
+    serviceHub.networkMapCache.allNodes.map { node -> node.legalIdentities.first() } - message.recipient - message.sender
+  subFlow(BroadcastTransactionFlow(it, broadcastToParties))
+}
+```
+
+At this point, after calling `BroadcastTransactionFlow`, the transaction will exist in the vault of each party passed to the flow. This includes the states contained within the transaction. Great, job done. You now know that Corda can broadcast transactions.
+
+There is one concern that needs to be addressed before you copy and paste this code and ship it straight to production. The states inside the broadcasted transactions can now be spent by any party who has received them. Therefore, if you wish to share transactions like this, then you need to put in the required safeguards to prevent organisations from spending states that don't really belong to them. Yes, that organisation might be a regulator and the chances of them doing anything dodgy is low. Yes, it might be possible to settle these issues outside of Corda. But, neither of these assumptions are ideal. To circumvent this, your contracts and flows must be designed to prevent organisations from spending states that are not theirs. I will cover this in a future post.
```

Comment:
They can be spent only if the contract code allows them to be spent by these other parties. Maybe this point can be made a bit clearer.

Reply:
I think the paragraph does put that message across + I'm finding it hard to rewrite this one.

I have added the following sentences that I think makes this 100% clear.

This responsibility falls onto you, the CorDapp developer. Not Corda. The flexibility Corda empowers you with must be controlled to prevent undesired outcomes.